pkg/v1/remote/options.go

```diff
@@ -78,26 +107,32 @@ func makeOptions(target authn.Resource, opts ...Option) (*options, error) {
 		o.auth = auth
 	}
 
-	// Wrap the transport in something that logs requests and responses.
-	// It's expensive to generate the dumps, so skip it if we're writing
-	// to nothing.
-	if logs.Enabled(logs.Debug) {
-		o.transport = transport.NewLogger(o.transport)
-	}
+	// transport.Wrapper is a signal that consumers are opt-ing into providing their own transport without any additional wrapping.
+	// This is to allow consumers full control over the transports logic, such as providing retry logic.
+	if _, ok := o.transport.(*transport.Wrapper); !ok {
```

Comment:
This is going to be tricky to maintain -- can you add a test that _doesn't_ get wrapped so we know this works?

Reply:
Finding a good place to test this wasn't a clear call.

Ended up writing a couple tests in the `MultiWrite` test:
-  Asserts that if a `transport.Wrapper` is used, then retrying should *not* occur. 
-  Asserts that if a `transport.Wrapper` is used, then User-Agent should not be wrapped..